addon/components/basic-dropdown-trigger.ts

```diff
@@ -107,24 +109,29 @@ export default class BasicDropdownTrigger extends Component<Args> {
     // This next three lines are stolen from hammertime. This prevents the default
     // behaviour of the touchend, but synthetically trigger a focus and a (delayed) click
     // to simulate natural behaviour.
-    let target = e.target as HTMLElement;
+    let target = (e.composedPath?.()[0] || e.target) as HTMLElement;
     if (target !== null) {
       target.focus();
     }
     setTimeout(function() {
-      if (!e.target) { return; }
+      if (!target) { return; }
       try {
         let event = document.createEvent('MouseEvents');
         event.initMouseEvent('click', true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
-        e.target.dispatchEvent(event);
-      } catch (e) {
+        target.dispatchEvent(event);
+      } catch (error) {
         event = new Event('click');
-        e.target.dispatchEvent(event);
+        target.dispatchEvent(event);
       }
     }, 0);
     e.preventDefault();
   }
 
+  @action
+  registerTriggerElement(triggerElement: HTMLElement): void {
+    this.args.onInsert?.(triggerElement);
```

Comment:
This passes a reference of trigger element back into the parent `basic-dropdown` component.

Essentially, it is a lightweight equivalent of https://github.com/lifeart/ember-ref-modifier . `ember-ref-modifier` is deprecated in favor of https://github.com/lifeart/ember-ref-bucket , but `ember-ref-bucket` does not work well for our case (the reference needs to be created in the child component and passed to parent).